runtime/js/telemetry.js

```diff
@@ -192,13 +223,28 @@ class Span {
     this.enter();
   }
 
+  get traceId() {
+    if (!this.#traceIdString) this.#traceIdString = bytesToHex(this.#traceId);
+    return this.#traceIdString;
+  }
+
+  get spanId() {
+    if (!this.#spanIdString) this.#spanIdString = bytesToHex(this.#spanId);
+    return this.#spanIdString;
+  }
+
+  get parentSpanId() {
+    if (!this.#parentSpanIdString && this.#parentSpanId) {
+      this.#parentSpanIdString = bytesToHex(this.#parentSpanId);
+    }
+    return this.#parentSpanIdString;
+  }
+
   // helper function to match otel js api
   spanContext() {
-    return {
-      traceId: this.traceId,
-      spanId: this.spanId,
-      traceFlags: this.traceFlags,
-    };
+    // No need to allocate a new object, we just need an object with the
```

Comment:
new object is intentional, `spanContext` can be held without holding the rest of the span for gc.

Reply:
will revert, should i make it an object with getters for the fields or, should it have fields?